### PR TITLE
New lff script and `org.lflang.cli` moved into `org.lflang` as subpackage 

### PR DESCRIPTION
**Pull request by cmnrd**
This PR refactors the org.lflang.cli package.

1. It includes the `org.lflang.cli` package as a subpackage in `org.lflang`. This closes https://github.com/lf-lang/lingua-franca/issues/1421.
2. It simplifies the gradle build such that a single jar (`org.lflang-0.3.1-SNAPSHOT.jar`) is created. This jar includes both the lfc and lff main classes. The main build task is now called `buildAll`. `buildLfc` is still available but shows a deprecation warning.
3. It introduces an `lff` script to invoke the lingua franca formatter. This closes https://github.com/lf-lang/lingua-franca/issues/1369.
4. It renames the build script to `build-lf-cli` and drops the `--run` option. Note that the old build script is still available, but prints a deprecation warning. The `--run` option was dropped because it would be unclear which command to run. Also the run tasks that we provide for gradle already implement this functionality. 

There are still a couple of unresolved issues though:
- [x] Fix and rename the `build-lfc` script. Should we name it `build-all`?
- [x] Add `lff.ps1` script for Windows. (@axmmisaka)
- [x] There is a test class [LfcIssueReportingTest.kt](https://github.com/lf-lang/lingua-franca/blob/f1c4e65912cfdd103ca4b3fdf7f9c1aa509e54b0/org.lflang.cli/test/kotlin/org/lflang/cli/tests/LfcIssueReportingTest.kt) in `org.lflang.cli.test`, but I was not able to move it to the new location and make it work with gradle. @oowekyala can you help with this?
- [x] We should update the developer documentation and use the `buildAll` task instead of `buildLfc`. Also `build-lf-cli` should be used instead of `build-lfc`. See https://github.com/lf-lang/website-lingua-franca/issues/106
- [x] Open an issue to keep track of the deprecated tools and remove them for next release. See https://github.com/lf-lang/lingua-franca/issues/1431
- [x] Update the `.github/scripts/package_lfc.sh` script (and rename to `package.sh`?)
- [x] Document changes required to release workflow https://github.com/lf-lang/release-tools/issues/16

### CODE REVIEW

**Comment by axmmisaka** [comment]
Haha, @cmnrd is faster than me on this…… I’ll review this PR then

**Comment by lhstrh** [comment]
Still some open tasks though!

**Comment by cmnrd** [comment]
@axmmisaka Do you have access to a Windows machine? You could fix the lfc.ps1 script and add a similar one for lff

**Comment by axmmisaka** [comment]
> @axmmisaka Do you have access to a Windows machine? You could fix the lfc.ps1 script and add a similar one for lff

Yes! I’ll take care of it

**Comment by axmmisaka** [comment]
Just as a reminder [this](https://github.com/lf-lang/lingua-franca/pull/1422/files#diff-4de59fd084b002242414d5ad66b2151f96a5ccf540845b5776686b60cb13f9b3R7) needs to be changed as well

**Comment by axmmisaka** [comment]
As of [c036e4c](https://github.com/lf-lang/lingua-franca/pull/1422/commits/c036e4cc9b0140660eb5cc0f94a8c77979adb4b5), `./gradlew buildAll` does not produce jar - `./gradlew assemble` is required.

**Comment by cmnrd** [comment]
@axmmisaka The jar is now produced in `org.lflang/build.libs/` and is called `org.lflang-0.3.1-SNAPSHOT.jar`. I just tested this on both Linux and Windows and it works fine for me. The main point of this PR is to remove `org.lflang.cli` and integrate its contents with `org.lflang`. Since the path (and name) of the jar changes, we also need to update the scripts. `org.lflang.cli` currently only contains a single test file, which I haven't figured out how to move yet (see 3rd item in the list in the description). The jar produced by `.\gradlew assemble` for `org.lflang.cli` is mostly empty for this reason.

File: .github/scripts/package-cli.sh [hunk 1]
```diff
@@ -24,13 +24,15 @@ mkdir -p "${outname}/lib/scripts"
...
-mv org.lflang.cli/build/libs/org.lflang.cli-*-lfc.jar "${outname}/lib/jars"
+mv org.lflang/build/libs/org.lflang-*.jar "${outname}/lib/jars"
 
 # copy the Bash scripts
 cp -a lib/scripts "${outname}/lib/"
 ln -s "../lib/scripts/launch.sh" "${outname}/bin/lfc"
+ln -s "../lib/scripts/launch.sh" "${outname}/bin/lff"
 # copy the PowerShell script
 cp bin/lfc.ps1 "${outname}/bin/lfc.ps1"
+# FIXME: Need to copy the lff.ps1 script once it exists
```
Comment (lhstrh) [inline .github/scripts/package-cli.sh:35]:
```suggestion
cp bin/lff.ps1 "${outname}/bin/lff.ps1"
```

File: org.lflang/build.gradle [hunk 2]
```diff
@@ -67,3 +69,67 @@ generateXtext.dependsOn(generateXtextLanguage)
...
+}
+
+task runLff(type: JavaExec) {
+    // builds and runs lff
+    description = "Build and run lff, use --args to pass arguments"
+    group = "application"
+    classpath = sourceSets.main.runtimeClasspath
+    mainClass = 'org.lflang.cli.Lff'
+    workingDir = '..'
+}
```
Comment (lhstrh) [inline org.lflang/build.gradle:135]:
```suggestion
}

```

**Review by lhstrh — APPROVED** [review]
:rocket: :rocket: :rocket:

**Comment by cmnrd** [comment]
As far as I am concerned this is ready to be merged! The only point that is still unclear to me is whether the changes here affect the release toolflow. @lhstrh can you comment on this and perhaps open an issue in the workflow repo?

**Comment by lhstrh** [comment]
> As far as I am concerned this is ready to be merged! The only point that is still unclear to me is whether the changes here affect the release toolflow. @lhstrh can you comment on this and perhaps open an issue in the workflow repo?

There might be some issue, but those will surface once I create a draft PR. This should not be a reason to hold up this PR and would only amount to changes required in the `release-tools` repo.

**Pull request merged**